lib/nerves_ssh/keys.ex

```diff
@@ -1,9 +1,19 @@
 defmodule NervesSSH.Keys do
   @moduledoc false
 
+  @creatable_host_key_algorithms [:"ssh-ed25519", :"rsa-sha2-256", :"rsa-sha2-512", :"ssh-rsa"]
+
   def host_key(algorithm, options) do
     # Delegate to system implementation for handling the host keys
-    :ssh_file.host_key(algorithm, options)
+    # If it fails, attempt to create the key and write it if
+    # it is a creatable algorithm. If not, return original error.
+    case :ssh_file.host_key(algorithm, options) do
```

Comment:
Going from the first comment, passing all of the algorithms down here would make it possible to check if a file was created with any of the supported algorithms first before creating one. That would fix an upgrade issue where if OTP added support for an algorithm, we might create a new key when a file already exists with one.